q2_feature_classifier/tests/test_classifier.py

```diff
@@ -102,31 +96,41 @@ def test_classify(self):
                        'fasta', rev_path)
         rev_reads = Artifact.import_data('FeatureData[Sequence]', rev_path)
 
-        fitter_name = _specific_fitters[0][0]
-        fitter = getattr(feature_classifier.methods,
-                         'fit_classifier_' + fitter_name)
-        classifier = fitter(reads, self.taxonomy).classifier
-
-        result = classify(reads, classifier)
+        result = classify(reads, self.classifier)
         fc = result.classification.view(pd.Series).to_dict()
-        result = classify(rev_reads, classifier)
+        result = classify(rev_reads, self.classifier)
         rc = result.classification.view(pd.Series).to_dict()
 
         for taxon in fc:
             self.assertEqual(fc[taxon], rc[taxon])
 
-        result = classify(reads, classifier,
-                          read_orientation='same')
+        result = classify(reads, self.classifier, read_orientation='same')
         fc = result.classification.view(pd.Series).to_dict()
-        result = classify(rev_reads, classifier,
+        result = classify(rev_reads, self.classifier,
                           read_orientation='reverse-complement')
         rc = result.classification.view(pd.Series).to_dict()
 
         for taxon in fc:
             self.assertEqual(fc[taxon], rc[taxon])
 
-        result = classify(reads, classifier, chunk_size=100, n_jobs=2)
+        result = classify(reads, self.classifier, chunk_size=100, n_jobs=2)
         cc = result.classification.view(pd.Series).to_dict()
 
         for taxon in fc:
             self.assertEqual(fc[taxon], cc[taxon])
+
+    def test_unassigned_taxa(self):
+        # classifications that don't meet the threshold should be "Unassigned"
+        classify = feature_classifier.methods.classify_sklearn
+        seq_path = self.get_data_path('se-dna-sequences.fasta')
+        reads = Artifact.import_data('FeatureData[Sequence]', seq_path)
+        result = classify(reads, self.classifier, confidence=1.)
+
+        ref = self.taxonomy.view(pd.Series).to_dict()
+        cls = result.classification.view(pd.Series).to_dict()
+
+        assert 'Unassigned' in cls.values()
+        for seq in reads.view(DNAIterator):
+            _id = seq.metadata['id']
```

Comment:
Oops, one more comment:
`_id` -> `id_`
stylistic again, but when you're avoiding clobbering defaults the underscore usually comes after, because it doesn't imply anything about information hiding.